mesonbuild/dependencies/dev.py

```diff
@@ -755,6 +755,56 @@ def _has_define(compiler: 'Compiler', dname: str, env: 'Environment') -> bool:
         defval, _ = compiler.get_define(dname, '', env, [], [])
         return defval is not None
 
+    # mstorsjo/msvc-wine scripts provides DIA SDK on linux.
+    # DIA SDK directory is stored at the root of downloaded MSVC files.
+    # We can find this path from the build environment of msvc-wine.
+    # msvc-wine can be used in two different ways:
+    @staticmethod
+    def _find_msvc_wine_path(compiler: 'Compiler') -> T.Optional[str]:
+        mlog.debug('looking for msvc-wine')
+
+        # One way adds <msvc>/bin/x64 to PATH. From there user executes 'cl' script,
+        # which sets up VS dev environment and invokes the real cl.exe via wine.
+        if compiler.id == 'msvc':
+            mlog.debug('- compiler is cl')
+            clpath = pathlib.Path(shutil.which('cl'))
+            if clpath.parent.name not in {'arm', 'arm64', 'x86', 'x64'}:
+                mlog.debug(f'- cl path does not look like msvc-wine: {clpath}')
+                return None
+            if clpath.parents[1].name != 'bin':
+                mlog.debug(f'- cl path does not look like msvc-wine: {clpath}')
+                return None
+            mlog.debug(f'- msvc-wine found at: {clpath.parents[2]}')
+            return str(clpath.parents[2])
+
+        # Other way requires calling <msvc-wine>/msvcenv-native.sh, which
+        # adds bunch of directories into INCLUDE and LIB envvars, and then
+        # using native clang-cl. (Meson currently doesn't support clang-cl
+        # on linux for cross-compiling, only clang with explicit MS-compatibility
+        # parameters)
+        # We'll look for '<msvc>/kits/<number>/include/<kit version>/winrt'
+        # in INCLUDE.
```

Comment:
Where does `INCLUDE` come from? It is an extremely general name so it is confusing.

Reply:
It comes from script `msvcenv-native.sh`, which is part of mstorsjo/msvc-wine. The script is used in same way as vcvars*.bat in MSVC. You source it in your shell before building with clang-cl. The script exports INCLUDE and LIB environment variables, pointing to MSVC headers/libs. Environment variables are then consumed by clang-cl.

See: https://github.com/mstorsjo/msvc-wine?tab=readme-ov-file#use-with-clanglld-in-msvc-mode